go/consensus/api/api.go

```diff
@@ -244,3 +249,11 @@ type GetSignerNonceRequest struct {
 	AccountAddress staking.Address `json:"account_address"`
 	Height         int64           `json:"height"`
 }
+
+// TransactionsWithResults is GetTransactionsWithResults response.
+//
+// Results[i] are the results of executing Transactions[i].
+type TransactionsWithResults struct {
+	Transactions [][]byte `json:"transactions"`
+	Results      [][]byte `json:"results"`
```

Comment:
I don't think we should expose raw, backend-dependent results as this seems like exposing internal details (e.g., things like Tendermint tags and their values used for events are internal details).

I think it would be better to define a consensus backend-agnostic results format and convert results to that format. The types mentioned below should live in the `transaction` package (`go/consensus/api/transaction`). For now it should have fields like:
```golang
type Result struct {
	Error Error `json:"error"`
}

func (r Result) IsSuccess() bool {
	return r.Error.Code == 0 // consider making this zero a constant in go/common/errors.go, also see line 43 there
}
```

Where `Error` is a type that supports our error encoding convention used in all the protocols (see `go/common/erorrs`), e.g.:
```golang
type Error struct {
	Module  string `json:"module,omitempty"`
	Code    uint32 `json:"code,omitempty"`
	Message string `json:"message,omitempty"`
}
```

We could also expose things like emitted events in addition to errors (e.g., by having an `Events []Event` field in `Result`), but this would require a union type like the following:
```golang
type Event struct {
	Staking  *staking.Event `json:"staking,omitempty"`
	Registry *registry.Event `json:"registry,omitempty"`
	RootHash *roothash.Event `json:"roothash,omitempty"`
}
```
And it would get populated by calling a lightweight version of `GetEvents` of individual services (it wouldn't be `GetEvents` directly as that fetches all transactions, it would require a slight refactor to allow processing of a single tx, e.g. similar to `onEventDataTx`).

Reply:
Thanks for the suggestions, please check the changes in`go/consensus/tendermint/staking/staking.go` before i continue impl. for other backends & clean things up.